### PR TITLE
Introduction to Netcat

### LINKED ISSUE #3117: Introduction to Netcat
**Introduction**
'Netcat' is a back-end tool or a program that reads and writes data across a network. It can serve a variety of functions including port scanning, transferring files, port listening and redirection, and also serve as a backdoor.
This tutorial is aimed to break down the complexity of Netcat and explain various roles of Netcat to the user. This article will also provide a 'tech tutorial' that will guide the user to understanding Netcat.

**Key Takeaways**
By the end of this tutorial, the user should be should have learned:
1. Learned various roles of Netcat
2. Install Netcat on both Kali Linux and Windows
3. Run major Netcat commands
4. Launch reverse shells(backdoor) with Netcat
5. Edit data across a network with Netcat

**Reference**
N/A

### PR DESCRIPTION
**Pull request by Kimosop-Wakanda**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3117 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by LinusMuema** [comment]
plagiarism check done :heavy_check_mark:

**Comment by Ericgacoki** [comment]
net
Hello @Kimosop-Wakanda, please rename this image to `hero.png` and remove the white spaces in your folder name.

File: content/articles/Introduction-to-Netcat/index.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
+### Introduction
+Netcat is a network analysis tool developed for security specialists and hackers to analyze a network in traffic.
+
+It is considered a `swiss-army knife` in information technology as it 
+
```
Comment (Ericgacoki) [inline content/articles/Introduction-to-Netcat/index.md:5]: Could you please explain the phrase `swiss-army-knife`.

File: content/articles/Introduction-to-Netcat/index.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+It is also a must-have tool for pen-tester since it is essential for network debugging and investigation.
+
+### Goals
+In this tutorial, we will mainly focus on the following:
+- [Executing general Netcat commands](#executing-commands-on-netcat)
+- [Modes of running Netcat](#modes-of-running-netcat)
+- [Uses of Netcat](#uses-of-netcat)
+
+### Executing commands on Netcat
+Netcat is only available in the command-line interface. You can access Netcat on the `command line` on Windows and `bash terminal` on Linux.
```
Comment (Ericgacoki) [inline content/articles/Introduction-to-Netcat/index.md:21]: Before getting into this, add the prerequisites and guide us on how we can install Netcat.

File: content/articles/Introduction-to-Netcat/index.md (hunk 1 above)
Comment (Kimosop-Wakanda) [reply content/articles/Introduction-to-Netcat/index.md]: Changes made @Ericgacoki

**Review by Ericgacoki — APPROVED** [review]

**Pull request merged**